### PR TITLE
ZDC data model change 

### PR DESCRIPTION
**Pull request by ddobrigk**
@cortesep @coppedis @jgrosseo up for discussion / approval! Thanks :-D

### CODE REVIEW

File: Detectors/AOD/src/AODProducerWorkflowSpec.cxx
Comment (aalkin) [inline Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: This is incorrect, it should be `"ZDC", 1`, the version for AOD tables is encoded in subspec. 
Update: the same goes for V0 table above, the only reason this works is that AOD producer's output is not used directly by tasks.

Comment (jgrosseo) [reply Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: OK, this is wrong for all of them. See v0, cascade, collision etc. So it seems to work but is not good of course...

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]:
```suggestion
namespace zdc_000 // needed to have the new dynamic columns with the same name
```

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]: Can we add comments for all those as well?

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h:1018]:
```suggestion
DECLARE_SOA_TABLE(Zdcs_000, "AOD", "ZDC", //! ZDC information, version 0
```

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]:
```suggestion
DECLARE_SOA_TABLE_VERSIONED(Zdcs_001, "AOD", "ZDC", 1, //! ZDC information, version 1, std::vector format
```

**Review by jgrosseo — COMMENTED** [review]
I have posted a few comments for cosmetic changes but the rest is fine with me

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (ddobrigk) [reply Framework/Core/include/Framework/AnalysisDataModel.h:1018]: thanks, taken :-)

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (ddobrigk) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: yes let me do that now ...

File: Detectors/AOD/src/AODProducerWorkflowSpec.cxx
Comment (ddobrigk) [reply Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: Shall we change this everywhere? Or we keep doing as is? What do you think?

Comment (aalkin) [reply Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: Probably better to do the rest in a separate PR, for bookkeeping reasons.

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (ddobrigk) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: Added, thanks!

File: Detectors/AOD/src/AODProducerWorkflowSpec.cxx
Comment (ddobrigk) [reply Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: You mean the rest or all of them? What do you think? :-D

Comment (aalkin) [reply Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: ZDC should be done correctly from the start here, I think. The rest can be fixed elsewhere so we do not mix them with ZDC-specific change.

Comment (coppedis) [reply Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: So we include in this PR all the fixes for the ZDC and then another request will be open for the remaining issues, correct?

Comment (ddobrigk) [reply Detectors/AOD/src/AODProducerWorkflowSpec.cxx]: Okay, the latest PR does as you suggested, @aalkin ! Can you please check if I did no do anything major wrong? (technically, though, it's in the part of the producer that is as-of-yet disabled, but I want to be sure it's ok already). If this is fine I would suggest we go ahead and merge so I can do a PR for the O2Physics part. Thanks!!!

**Comment by ddobrigk** [comment]
Marked as ready to review to trigger all checks already (while also waiting for @aalkin 's green light). Thanks!

**Review by jgrosseo — DISMISSED** [review]

**Comment by jgrosseo** [comment]
Is the PR for the converter already there (as draft)?

**Comment by ddobrigk** [comment]
> Is the PR for the converter already there (as draft)?

I don't think I PR-ed it but I have it in [my fork here](https://github.com/AliceO2Group/O2Physics/compare/master...ddobrigk:O2Physics:zdcdevel). I will do a PR as soon as this is merged; anyway now I see some CI errors that I have to look into for O2Physics. More soon...

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (aalkin) [inline Framework/Core/include/Framework/AnalysisDataModel.h]: Please use the same namespace, the versioning should only be done at the table level, otherwise you will have to rename the namespace later.

Comment (ddobrigk) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: Hi Anton, as explained in the comment inside the code, I need to have different namespaces to have different types of columns with the same names: we are swapping out a regular column with a dynamic column in the new version of the table, and these have the same exact name. This cannot be done keeping the same namespace, unfortunately...

Comment (aalkin) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: Since the columns will have different type, they may as well have a different name. In fact, you can rename old columns (only the type names, not the getters). This will be transparent for the users, since they are only using the getters.

Comment (ddobrigk) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: The latest commit should take care: it uses only the `zdc` namespace but changes the column _names_ for the dynamic columns while keeping the same _getters_. Can you please take a look @aalkin ? Thanks a lot!

**Comment by ddobrigk** [comment]
Okay, so the latest CI errors seem unrelated and the code does compile in the regular tests... Now I would say we just need a confirmation with @aalkin to see if everything is in order and then we can merge and proceed to the converter part: I will check if there is maybe any small update that may be required on that side and prepare a PR. Thanks a lot, everyone!!

**Comment by aalkin** [comment]
I have no additional comments, @ddobrigk. Although you may want to produce some new format data files first and test them with O2Physics.

**Comment by ddobrigk** [comment]
> I have no additional comments, @ddobrigk. Although you may want to produce some new format data files first and test them with O2Physics.

Yes, this is a good idea. We did it already with the previous version, but we can redo it. For now, because the data model changes will only kick in after the `define` is uncommented (or, better, the code cleaned up from the previous versions), I suggest we merge and then do the test simultaneously; then the necessary condition to move to the new table in O2 (after also the O2Physics converter PR is merged) is to have the latest code tested.

**Review by jgrosseo — APPROVED** [review]

**Comment by jgrosseo** [comment]
I think the CI is unrelated. @ktf can you confirm and then merge?

**Comment by ddobrigk** [comment]
All tests are successful and we should really move on - @shahor02 or @ktf , could you please help us out? I think this is ready to go and agreed upon. Thanks!

**Comment by shahor02** [comment]
I'll run the fullCI locally, will merge if ok.

**Comment by shahor02** [comment]
passed fullCI locally, merging

**Comment by ddobrigk** [comment]
Thank you very much @shahor02 ! I will prepare the O2Physics PR shortly. Have a nice easter break!

**Pull request merged**
